kubee2etests/runners/request_runners.py

```diff
@@ -36,3 +40,41 @@ def run(self):
 
     def finish(self):
         self.deployment.change_cfg_map(self.cfgmap.name, report=False)
+
+
+class DNSRequestRunner():
+    def __init__(self,namespace,service,deployment):
+        super().__init__()
+        self.qname = TEST_DNS_QUERY_NAME
+
+    def add_error(self, err):
+        error_list = [error[0] for error in self.errors]
+        try:
+            idx = error_list.index(err)
+            self.errors[idx] = (err, self.errors[idx][1] + 1)
+        except ValueError:
+            self.errors.append((err, 1))
+
+    def incr_dns_count_metric(self,result):
+        """
+        Helper method which increments the dns request count metric.
+        Args:
+            result: string of what happened - healthy, nxdomain etc
+        Returns: None, increments the statsd dns count metric
+        """
+        result_data = {"result": result}
+        STATSD_CLIENT.incr(DNS_COUNT_METRIC_NAME % result_data)
+
+    def exec(self):
+        try:
+            dns.resolver.query(self.qname)
+            result="healthy"
+            LOGGER.info("DNS healthy")
+        except Exception as ex:
+            result=type(ex).__name__.lower()
+            LOGGER.error("Querying %s failed. %s", self.qname, ex)
+            self.add_error(ex)
+        self.incr_dns_count_metric(result)
+        
```

Comment:
Hmm, think we also need to add something which sends the errors to the flask frontend (think ApiMixin does this)